### ISSUE TITLE
Metrics request canceled while waiting for connection

### ISSUE THREAD

**Issue by erikwilson** [rancher/k3s#450]
**Describe the bug**
Unable to retrieve metrics:
```
[ 2019-05-06 16:58:08 ]
root🐮k3s-1:~$ kubectl get pods -A
NAMESPACE     NAME                              READY   STATUS      RESTARTS   AGE
kube-system   coredns-857cdbd8b4-qshzr          1/1     Running     0          3m55s
kube-system   helm-install-traefik-88kc5        0/1     Completed   0          3m55s
kube-system   metrics-server-675d76bcdb-sjczg   1/1     Running     0          3m55s
kube-system   svclb-traefik-86cfd577d6-2prrw    2/2     Running     0          3m6s
kube-system   svclb-traefik-86cfd577d6-4qnhh    2/2     Running     0          3m6s
kube-system   traefik-55bd9646fc-nthlj          1/1     Running     0          3m6s
[ 2019-05-06 16:58:12 ]
root🐮k3s-1:~$ kubectl top nodes
Error from server (ServiceUnavailable): the server is currently unable to handle the request (get nodes.metrics.k8s.io)
[ 2019-05-06 16:59:40 ]
root🐮k3s-1:~$ kubectl describe -n kube-system pod/metrics-server-675d76bcdb-sjczg | grep "^Node:"
Node:               k3s-2/10.135.135.102
```
From logs:
```
E0505 19:32:22.720216    7381 available_controller.go:353] v1beta1.metrics.k8s.io failed with: Get https://10.43.157.116:443: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

**To Reproduce**
Create multi-node cluster where metrics-server is deployed to a node other than the server (delete pod if lands on first node)

**Expected behavior**
Functional kubectl top

**Comment by galal-hussein** [comment]
I am not able to reproduce the issue:
- provision two ec2 instances on aws
- run k3s server on one and agent on another
- start metrics server from deployments recipes in the repo
- make sure metrics server pod landed on the agent node

Results:
```
root@ip-172-31-39-97:~/k3s# k3s kubectl top pods -n kube-system
NAME                              CPU(cores)   MEMORY(bytes)   
coredns-695688789-jj5r5           3m           16Mi            
helm-install-traefik-vds5v        0m           0Mi             
metrics-server-57479c7997-b59xp   2m           11Mi            
svclb-traefik-grggp               0m           2Mi             
svclb-traefik-s2rnq               0m           2Mi             
traefik-55bd9646fc-96khb          3m           18Mi
```
Looks like the problem was a networking issue between the nodes.

**Comment by galal-hussein** [comment]
k3s version: v0.7.0-rc8

steps to validate:

1- create 2 nodes on vagrant
2- start k3s server on the first node with the flannel-iface pointing to the secondary interface:
```
curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v0.7.0-rc8 sh -s - server --flannel-iface=enp0s8
```
3- start k3s agent on the second node with flannel-iface pointing to the secondary interface
```
curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION=v0.7.0-rc8 K3S_TOKEN=xxxx K3S_URL=https://192.168.1.5:6443 sh -s - --flannel-iface=enp0s8
```
4- deploy metrics server


I was able to see the metrics pod starting correctly and gathering metrics for pods and nodes correctly:
```
# kubectl top nodes
NAME       CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%   
agent01    16m          1%     550Mi           27%       
server01   46m          4%     442Mi           22%
```
```
# kubectl top pods --all-namespaces
NAMESPACE     NAME                              CPU(cores)   MEMORY(bytes)   
kube-system   coredns-b7464766c-fvdj6           2m           14Mi            
kube-system   metrics-server-7cffff65bc-g6knw   1m           11Mi            
kube-system   svclb-traefik-96z8k               0m           1Mi             
kube-system   svclb-traefik-w4vc4               0m           2Mi             
kube-system   traefik-56688c4464-td79w          2m           16Mi 
```

**Comment by shibumi** [comment]
@erikwilson did you solve this issue? I have the same issue with it on rancher with a cluster on-premise.

**Comment by brandond** [comment]
@shibumi this issue has been closed and appears to have been fixed in v0.7.0 almost a year ago. If you're still experiencing this, please open a new issue after confirming that you're no longer running an affected release.
